frontend/src/pages/RecipientRecord/pages/CommunicationLogForm/index.js

```diff
@@ -0,0 +1,342 @@
+import React, {
+  useEffect,
+  useState,
+  useContext,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { Helmet } from 'react-helmet';
+import { Alert, Grid } from '@trussworks/react-uswds';
+import { useHistory, Redirect } from 'react-router-dom';
+import { FormProvider, useForm } from 'react-hook-form';
+import useHookFormPageState from '../../../../hooks/useHookFormPageState';
+import { defaultValues, formatCommunicationLogUrl, recipientRecordRootUrl } from './constants';
+import NetworkContext, { isOnlineMode } from '../../../../NetworkContext';
+import UserContext from '../../../../UserContext';
+import Navigator from '../../../../components/Navigator';
+import BackLink from '../../../../components/BackLink';
+import pages from './pages';
+import AppLoadingContext from '../../../../AppLoadingContext';
+import {
+  updateCommunicationLogById,
+  createCommunicationLogByRecipientId,
+  getCommunicationLogById,
+} from '../../../../fetchers/communicationLog';
+
+/**
+ * this is just a simple handler to "flatten"
+ * the JSON column data into the form
+ *
+ * @param {fn} reset this is the hookForm.reset function (pass it a new set of values and it
+ *  replaces the form with those values; it also calls the standard form.reset event
+ * @param {*} event - not an HTML event, but the event object from the database, which has some
+ * information stored at the top level of the object, and some stored in a data column
+ */
+const resetFormData = (reset, updatedLog) => {
+  const {
+    data,
+    updatedAt,
+    ...fields
+  } = updatedLog;
+
+  const form = {
+    ...defaultValues,
+    ...data,
+    ...fields,
+  };
+
+  reset(form);
+};
+
+const shouldFetch = (
+  communicationLogId,
+  regionId,
+  recipientId,
+  reportFetched,
+  isAppLoading,
+  currentPage,
+) => {
+  if (
+    !currentPage
+    || !communicationLogId
+    || !regionId
+    || !recipientId
+    || communicationLogId === 'new'
+    || reportFetched
+    || isAppLoading) {
```

Comment:
I assume we need to check all of these for all cases. Can we comment on what we are checking and why.